File: test_archives/py_tests/batch2/test_function8.py
from functions.function8 import function8
import pytest

def test_function8_1():
    assert function8("hello", [1, 2, 3], True, 4, 5.2) == "hello4"
    
def test_function8_2():
    assert function8("world", [4, 5, 6], False, 7, 8.3) == "[4, 5, 6]8.3"

File: test_archives/cpp_tests/batch2/test_function8.cpp
#include "../catch.hpp"
#include "../functions/function8.cpp"

TEST_CASE( "Test function8_1", "[function8]" ) {
    REQUIRE( function8("hello", {1, 2, 3}, true, 4, 5.2) == "hello4" );
}

TEST_CASE( "Test function8_2", "[function8]" ) {
    REQUIRE( function8("world", {4, 5, 6}, false, 7, 8.3) == "[4, 5, 6]8.3" );
}